ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,408 @@
+variables constrained to belong in a vectorized manifold `VectorizedManifold`.
+"""
+function MOI.supports_add_constrained_variables(::Optimizer, ::Type{<:VectorizedManifold})
+    return true
+end
+
+"""
+    MOI.add_constrained_variables(model::Optimizer, set::VectorizedManifold)
+
+Add `MOI.dimension(set)` variables constrained in `set` and return the list
+of variable indices that can be used to reference them as well a constraint
+index for the constraint enforcing the membership of the variables in the
+`VectorizedManifold` `set`.
+"""
+function MOI.add_constrained_variables(model::Optimizer, set::VectorizedManifold)
+    F = MOI.VectorOfVariables
+    if !isnothing(model.manifold)
+        throw(
+            AddConstraintNotAllowed{F,typeof(set)}(
+                "Only one manifold allowed, variables in `$(model.manifold)` have already been added.",
+            ),
+        )
+    end
+    model.manifold = set.manifold
+    model.problem = nothing
+    model.state = nothing
+    n = MOI.dimension(set)
+    v = MOI.VariableIndex.(1:n)
+    for _ in 1:n
+        push!(model.variable_primal_start, nothing)
+    end
+    return v, MOI.ConstraintIndex{F,typeof(set)}(1)
+end
+
+"""
+    MOI.is_valid(model::Optimizer, vi::MOI.VariableIndex)
+
+Return whether `vi` is a valid variable index.
+"""
+function MOI.is_valid(model::Optimizer, vi::MOI.VariableIndex)
+    return !isnothing(model.manifold) &&
+           1 <= vi.value <= MOI.dimension(VectorizedManifold(model.manifold))
+end
+
+"""
+    MOI.get(model::Optimizer, ::MOI.NumberOfVariables)
+
+Return the number of variables added in the model, this corresponds
+to the [`MOI.dimension`](@ref) of the `VectorizedManifold`.
+"""
+function MOI.get(model::Optimizer, ::MOI.NumberOfVariables)
+    if isnothing(model.manifold)
+        return 0
+    else
+        return MOI.dimension(VectorizedManifold(model.manifold))
+    end
+end
+
+"""
+    MOI.supports(::Manopt.JuMP_Optimizer, attr::MOI.RawOptimizerAttribute)
+
+Return `true` indicating that `Manopt.JuMP_Optimizer` supports starting values
+for the variables.
+"""
+function MOI.supports(::Optimizer, ::MOI.VariablePrimalStart, ::Type{MOI.VariableIndex})
+    return true
+end
+
+"""
+    function MOI.set(
+        model::Optimizer,
+        ::MOI.VariablePrimalStart,
+        vi::MOI.VariableIndex,
+        value::Union{Real,Nothing},
+    )
+
+Set the starting value of the variable of index `vi` to `value`. Note that if
+`value` is `nothing` then it essentially unset any previous starting values set
+and hence `MOI.optimize!` unless another starting value is set.
+"""
+function MOI.set(
+    model::Optimizer,
+    ::MOI.VariablePrimalStart,
+    vi::MOI.VariableIndex,
+    value::Union{Real,Nothing},
+)
+    MOI.throw_if_not_valid(model, vi)
+    model.variable_primal_start[vi.value] = value
+    model.state = nothing
+    return nothing
+end
+
+function MOI.supports(::Optimizer, ::Union{MOI.ObjectiveSense,MOI.ObjectiveFunction})
```

Comment:
Similar to above – sure this is nearly readable what it is meant to indicate – but I am for example not 100% sure what `ObjectiveSense` refers to – for me all objectives make sense 😬